src/Kunstmaan/NodeBundle/Router/SlugRouter.php

```diff
@@ -36,29 +49,45 @@ class SlugRouter implements RouterInterface
     /** @var UrlGenerator */
     protected $urlGenerator;
 
-    /** @var ContainerInterface */
+    /**
+     * @var ContainerInterface
+     * @deprecated in KunstmaanNodeBundle 5.1 and will be removed in KunstmaanNodeBundle 6.0.
+     */
     protected $container;
 
     /** @var string */
     protected $slugPattern;
 
-    /** @var DomainConfigurationInterface */
-    protected $domainConfiguration;
-
-    /** @var string */
-    protected $adminKey;
-
     /**
      * The constructor for this service
      *
      * @param ContainerInterface $container
      */
-    public function __construct($container)
+    public function __construct(
+        /* DomainConfigurationInterface */ $domainConfiguration,
+        RequestStack $requestStack = null,
+        EntityManagerInterface $em = null,
+        $adminKey = null
+    )
     {
-        $this->container = $container;
         $this->slugPattern = "[a-zA-Z0-9\-_\/]*";
-        $this->domainConfiguration = $container->get('kunstmaan_admin.domain_configuration');
-        $this->adminKey            = $container->getParameter('kunstmaan_admin.admin_prefix');
+
+        if ($domainConfiguration instanceof ContainerInterface) {
+            @trigger_error('Container injection and the usage of the container is deprecated in KunstmaanNodeBundle 5.1 and will be removed in KunstmaanNodeBundle 6.0.', E_USER_DEPRECATED);
+
+            $this->container = $domainConfiguration;
+            $this->domainConfiguration = $domainConfiguration->get('kunstmaan_admin.domain_configuration');
```

Comment:
Maybe use $this->container->get here? Looks weird getting stuff from domainConfiguration 😄

Reply:
Indeed that will be better/more clear! Fixed!